File: app/src/main/java/ua/zt/mezon/graphomania/fsmandstrategydemo/stratm/IStrategyState.java
package ua.zt.mezon.graphomania.fsmandstrategydemo.stratm;


import ua.zt.mezon.graphomania.fsmandstrategydemo.utils.fsm.FSMAction;

/**
 * Created by NickZT on 01.02.2019.
 */
public
interface IStrategyState {


    String getStateDesc();

    void addAction(FSMAction fsmAction);

    void executeAction();
}

File: app/src/main/java/ua/zt/mezon/graphomania/fsmandstrategydemo/utils/stratm/IStrategyState.kt
package ua.zt.mezon.graphomania.fsmandstrategydemo.utils.stratm

import ua.zt.mezon.graphomania.fsmandstrategydemo.utils.fsm.FSMAction

/**
 * Created by NickZT on 01.02.2019.
 */
interface IStrategyState {
    val stateDesc: String
    fun addAction(fsmAction: FSMAction)
    fun executeAction()
}